src/app/likes/likes.component.spec.ts

```diff
@@ -125,44 +127,129 @@ describe('LikesComponent', () => {
 
-    const anchor_list_items = fixture.debugElement.queryAll(By.css('a'));
+    const mat_grid_tile_items = fixture.debugElement.queryAll(
+      By.css('.mat-grid-tile')
+    );
 
-    expect(anchor_list_items.length).toEqual(likes.length);
+    expect(mat_grid_tile_items.length).toEqual(likes.length);
+
+    const mat_grid_tile_list_content = mat_grid_tile_items.map(list_item => {
+      return {
+        song_art_image_url: String(
+          list_item.nativeElement.style.backgroundImage.replace(
+            /url\(\"(.*)\"\)/gm,
+            '$1'
+          )
+        ),
+      };
+    });
+
+    const expected_content = likes.map(({song_art_image_url}) => ({
+      song_art_image_url: String(song_art_image_url),
+    }));
+
+    expect(mat_grid_tile_list_content).toEqual(expected_content);
+  }));
+
+  it('should display the header and footer for the hovered like', async(() => {
+    expect(component.hoveredLikeId).toEqual(null);
+
+    fixture.detectChanges();
+
+    fixture.debugElement
+      .queryAll(By.css('.mat-grid-tile'))[0]
+      .nativeElement.dispatchEvent(new Event('mouseover'));
+
+    fixture.detectChanges();
+
+    expect(component.hoveredLikeId).toEqual(likeId);
 
-    const anchor_list_content = anchor_list_items.map(list_item => {
-      const song_artist = /(.*) by (.*)/gm.exec(
-        list_item.nativeElement.getAttribute('title')
+    for (let margin of ['header', 'footer']) {
+      const hovered_mat_grid_tile_margin = fixture.debugElement.queryAll(
+        By.css(`.mat-grid-tile-${margin}`)
       );
-      if (song_artist && song_artist.length == 3) {
-        return {
-          artist: String(song_artist[2]),
-          title: String(song_artist[1]),
-          song_art_image_url: String(
-            list_item.nativeElement
-              .getElementsByClassName('mat-grid-tile')[0]
-              .style.backgroundImage.replace(/url\(\"(.*)\"\)/gm, '$1')
-          ),
-          url: String(list_item.nativeElement.getAttribute('href')),
-        };
-      }
-    });
 
-    const expected_content = likes.map(
-      ({artist, title, song_art_image_url, url}) => ({
-        artist: String(artist),
-        title: String(title),
-        song_art_image_url: String(song_art_image_url),
-        url: String(url),
-      })
-    );
+      expect(hovered_mat_grid_tile_margin.length).toEqual(1);
+    }
+
+    const hovered_mat_grid_tile_content = fixture.debugElement
+      .queryAll(By.css('.mat-grid-tile'))
+      .slice(0, 1)
+      .map(tile => {
+        const song_artist = /(.*) by (.*)/gm.exec(
+          tile.nativeElement.getElementsByClassName('mat-grid-tile-header')[0]
+            .innerText
+        );
+        if (song_artist && song_artist.length == 3) {
+          return {
+            artist: String(song_artist[2]),
+            title: String(song_artist[1]),
+            url: String(tile.nativeElement.getElementsByTagName('a')[0].href),
+          };
+        }
```

Comment:
```suggestion
        if (!song_artist || song_artist.length !== 3) return;
        return {
          artist: String(song_artist[2]),
          title: String(song_artist[1]),
          url: String(tile.nativeElement.getElementsByTagName('a')[0].href),
        };
```

- Use guard clauses to reduce nesting
- Use === and similar for comparison, not ==